common/v2/services/Store/Account/AccountProvider.tsx

```diff
@@ -32,6 +34,23 @@ export class AccountProvider extends Component {
       service.updateAccount(uuid, accountData);
       this.getAccounts();
     },
+    addNewTransactionToAccount: (accountData, newTransaction) => {
+      const existingTransaction = accountData.transactions.find(
+        tx => tx.hash === newTransaction.hash
+      );
+      delete newTransaction.network;
+      if (existingTransaction) {
+        const newTransactionSet = accountData.transactions.filter(
+          transaction => transaction.hash !== newTransaction.hash
+        );
+        newTransactionSet.push(newTransaction);
+        accountData.transactions = [...newTransactionSet];
```

Comment:
prefer the declarative form:
```
   accountData.transactions = [...newTransactionSet, newTransaction];
```

NB. Instead of telling the machine "append this to the end of the list" (imperative) we write "this comes at the end" (declarative).